lms/djangoapps/instructor/views/api.py

```diff
@@ -3313,6 +3340,47 @@ def certificate_invalidation_view(request, course_id):
         return JsonResponse({}, status=204)
 
 
+@transaction.non_atomic_requests
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_global_staff
+@require_http_methods(['POST'])
+def update_cohort_assignment(request, course_id):
+    """
+    Api endpoint for Cohort Managers updating.
+    """
+    course_id = CourseKey.from_string(course_id)
+    course = get_course_with_access(
+        request.user, 'instructor', course_id, depth=None
+    )
+
+    is_assignment = json.loads(request.POST.get('is_assignment').lower())
+    cohort_id = int(request.POST.get('cohort_id'))
+    user =get_student_from_identifier(request.POST.get('unique_student_identifier'))
+
+    #Note (yura.braiko@gmail.com): if cohort id is `-1` it equal to `All` cohort.
+    if cohort_id == -1:
+        migrate_cohort_settings(course)
+
+        cohorts_list = CourseUserGroup.objects.filter(
+            course_id=course.location.course_key,
+            group_type=CourseUserGroup.COHORT
+        ).values_list('id')
+
+        cohort_for_change = [item[0] for item in cohorts_list]
+    else:
+        cohort_for_change = [cohort_id]
+
+    if is_assignment:
+        cohort_to_incert = [CohortAssigment(cohort_id=cohort, user=user) for cohort in cohort_for_change]
+        CohortAssigment.objects.bulk_create(cohort_to_incert)
+    else:
+        CohortAssigment.objects.filter(cohort__id__in=cohort_for_change, user=user).delete()
+
+    # Note(yura.braiko@raccoongang.com): return empty success response.
+    return JsonResponse({})
```

Comment:
Maybe we need add successful message?

Reply:
As we do not process it, I don't think so.